Datastore/src/Query/Aggregation.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * Copyright 2023 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Datastore\Query;
+
+/**
+ * Represents Count Aggregation properties.
+ *
+ * Example:
+ * ```
+ * $count = Aggregation::count();
+ * $count->alias('count');
+ * $count->limit(100);
+ *
+ * echo json_encode($count->getProps());
+ * ```
+ */
+class Aggregation
+{
+    /**
+     * Default placeholder for all count aggregation props.
+     */
+    const TYPE_COUNT = 'count';
+
+    /**
+     * @var array Properties for an aggregation query.
+     */
+    private $props = [];
```

Comment:
Reminder that we can start using typed properties!
```suggestion
    private array $props = [];
```

I personally think it'd be good to get in the habit of defining them when we can on new classes, but that's not required for this PR!